src/geo/spatial.clj

```diff
@@ -94,23 +92,36 @@
   (>= (.getWidth (.getEnvelopeInternal jts-geom))
       180))
 
+(defn shape->geom
```

Comment:
I'm wondering if this can be handled as part of the protocol, since that's really the ideal mechanism for doing type-based dispatch like this. It might need to be added as new cases in the `Shapelike` protocol, but it seems like these are really implementations of the `to-jts` portion of that protocol for the `RectangleImpl`, `JtsGeometry`, `PointImpl`, and `JtsPoint` classes

Reply:
I don't know why I didn't think of this. This will mean removing `(to-jts)` from Shape, and only adding it to the types currently listed in `shape->geom` (excluding circles, basically), but it should definitely be faster.